src/app/shared/model/base.model.ts

```diff
@@ -137,6 +142,11 @@ export abstract class BaseModel {
     this.unstore();
     this.lastStored = null;
     this.isDestroy = false;
+    if (!this.doc && this.original) {
```

Comment:
This is a bit of a typescript gotcha.  All the default attributes at the top of the class (like `public thing = 'default-value-here';`) are actually evaluated as part of the constructor.  So for unsaved models, I need to recover those values from the `this.original` hash.